lib/api/controllers/document.js

```diff
@@ -310,12 +311,16 @@ class DocumentController extends NativeController {
           { refresh, userId, retryOnConflict });
       })
       .then(_response => {
-        // response: { _id, _version }
+        // response: { _id, _version, _source }
         response = _response;
-
-        return this.kuzzle.notifier.notifyDocumentUpdate(modifiedRequest);
+        return this.kuzzle.notifier.notifyDocumentUpdate(modifiedRequest, response);
       })
-      .then(() => response);
+      .then(() => {
+        if (!source) {
+          delete response._source;
```

Comment:
It's better to return a new object without the `_source` property than deleting a property in the existing `response`.  
The `delete` keyword will trigger a deoptimization of the code